docs/topics/dokkatoo-migration.md

```diff
@@ -0,0 +1,376 @@
+
+2. In the project's `gradle.properties` file, set the following opt-in flag with helpers to activate the new plugin version:
+
+   ```kotlin
+   org.jetbrains.dokka.experimental.gradle.pluginMode=V2EnabledWithHelpers
+   ```
+
+   If your project does not have a `gradle.properties` file, create one in the root directory of your project.
+
+3. Sync your project with Gradle to ensure the new plugin is properly applied:
+
+   * If you use IntelliJ IDEA, click the **Reload** button from the Gradle tool window.
+   * If you use Android Studio, select **File** | **Sync Project with Gradle Files**.
+
+## Migrate your project
+
+After updating the Dokka Gradle plugin to v2, follow the migration steps applicable to your project.
+
+### Adjust configuration options
+
+DGP v2 introduces some changes in the Gradle configuration options. In the `build.gradle.kts` file, adjust the configuration 
+options according to your project setup: 
+
+* **New top-level DSL configuration:** Replace the old configuration syntax with the 
+  new top-level `dokka {}` DSL configuration. For example:
+
+  Previous configuration:
+
+  ```kotlin
+  tasks.withType<DokkaTask>().configureEach {
+      dokkaSourceSets {
+          named("main") {
+              moduleName.set("Project Name")
+              includes.from("README.md")
+              sourceLink {
+                  localDirectory.set(file("src/main/kotlin"))
+                  remoteUrl.set(URL("https://example.com/src"))
+                  remoteLineSuffix.set("#L")
+              }
+          }
+      }
+  }
+
+  tasks.dokkaHtml {
+      pluginConfiguration<DokkaBase, DokkaBaseConfiguration> {
+          customStyleSheets.set(listOf("styles.css"))
+          customAssets.set(listOf("logo.png"))
+          footerMessage.set("(c) Your Company")
+      }
+  }
+  ```
+ 
+  New configuration:
+
+  ```kotlin
+  dokka {
+      moduleName = "Project Name"
+      dokkaSourceSets.main {
+          includes.from("README.md")
+          sourceLink {
+              localDirectory = file("src/main/kotlin")
+              remoteUrl = URL("https://example.com/src")
+              remoteLineSuffix = "#L"
+          }
+      }
+      pluginsConfiguration.html {
+          customStyleSheets.from("styles.css")
+          customAssets.from("logo.png")
+          footerMessage = "(c) Your Company"
+      }
+  }
+  ```
+
+* **Visibility settings:**  The `documentedVisibilities` property has changed from `Visibility.PUBLIC` to `VisibilityModifier.Public`.
+
+  Previous configuration:
+
+  ```kotlin
+  import org.jetbrains.dokka.DokkaConfiguration.Visibility
+  ...
+  documentedVisibilities.set(
+      setOf(Visibility.PUBLIC)
+  )  
+  ```
+
+  New configuration:
+
+  ```kotlin
+  import org.jetbrains.dokka.gradle.engine.parameters.VisibilityModifier
+  ...
+  documentedVisibilities.set(
+      setOf(VisibilityModifier.Public)
+  )
+  ```
```

Comment:
```suggestion
  ```kotlin
  import org.jetbrains.dokka.DokkaConfiguration.Visibility
  // ...
  documentedVisibilities.set(
      setOf(Visibility.PUBLIC)
  )  
  ```

  New configuration:

  ```kotlin
  import org.jetbrains.dokka.gradle.engine.parameters.VisibilityModifier
  // ...
  documentedVisibilities.set(
      setOf(VisibilityModifier.Public)
  )
  ```
```

Let's make the `...` comments so the code is easier to copy and paste.